### ISSUE TITLE
Increased allocations in latest released

### ISSUE THREAD

**Issue by charleskawczynski** [CliMA/ClimaCore.jl#881]
In the latest release (I suspect #829 is responsible), we found in [TC#1261](https://github.com/CliMA/TurbulenceConvection.jl/pull/1261) that allocations in initialization increased from 25GB to 40GB:

Before:

```
[ Info: The simulation has completed.
  | 258.535694 seconds (483.68 M allocations: 25.021 GiB, 5.09% gc time, 94.81% compilation time)
```

After
```
[ Info: The simulation has completed.
  | 312.802070 seconds (858.13 M allocations: 40.089 GiB, 7.48% gc time, 96.01% compilation time)
```

We should verify which set of changes did this and fix / add some sort of tests.

**Comment by charleskawczynski** [comment]
We've fixed a handful of the ambiguities, I think the codegen is the remaining culprit I can think of

**Comment by charleskawczynski** [comment]
Closed by #1113.
